src/stratis_cli/_actions/_top.py

```diff
@@ -172,27 +172,31 @@ def add_data_devices(namespace):
             .search(managed_objects)
         )
 
-        cache = [
-            MODev(info).Devnode()
+        blockdevs = frozenset(namespace.blockdevs)
+
+        cache = frozenset(
+            str(MODev(info).Devnode())
             for (_, info) in devs(props={"Tier": BlockDevTiers.Cache}).search(
                 managed_objects
             )
-        ]
-        already_cache = [dev for dev in namespace.blockdevs if dev in cache]
-        if already_cache != []:
-            raise StratisInUseError(already_cache, BlockDevTiers.Data)
+        )
+        already_cache = blockdevs.intersection(cache)
+        if list(already_cache) != []:
```

Comment:
Don't convert it to a list...just compare with empty frozenset.

Reply:
Resolved.